nym-vpn-core/crates/nym-gateway-directory/src/entries/gateway.rs

```diff
@@ -37,6 +44,20 @@ impl Gateway {
     pub fn has_ipr_address(&self) -> bool {
         self.ipr_address.is_some()
     }
+
+    pub fn clients_address_no_tls(&self) -> Option<String> {
+        match (&self.host, &self.clients_ws_port) {
+            (Some(host), Some(port)) => Some(format!("ws://{}:{}", host, port)),
```

Comment:
for example with this, if you have `host`, you **must** have the `ws` port. there are no situations under which you can have one but not the other